pkg/oc/cli/admin/migrate/scc/scc.go

```diff
@@ -0,0 +1,357 @@
+package scc
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/origin/pkg/oc/cli/admin/migrate"
+	securityapi "github.com/openshift/origin/pkg/security/apis/security"
+	policy "k8s.io/api/policy/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	rbacv1helper "k8s.io/kubernetes/pkg/apis/rbac/v1"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource"
+)
+
+// clusterRoleAndBindingNamePrefix is a prefix that each auto-generated ClusterRole and ClusterRoleBinding have.
+// It simplifies their identification.
+const clusterRoleAndBindingNamePrefix = "psp:"
+
+// pspPriorityAnnotationKey is a name of the annotation that could be present on PodSecurityPolicy and specifies its priority.
+// It should be a number. An empty value means 0 priority. Higher values means that the policy has higher priority.
+// This is an equivalent of the SCC.Priority field but applies to PSP and it's required for seamless transition.
+const pspPriorityAnnotationKey = "security.openshift.io/psp-priority"
+
+var (
+	internalMigrateSCCShort = "Converts SCCs to similar PSPs"
+	internalMigrateSCCLong  = templates.LongDesc(`
+		Converts SecurityContextConstraints to the equivalent PodSecurityPolicy objects.
+		For each SCC it also creates a ClusterRole with the similar name and ClusterRoleBinding
+		for every user or group that SCC has. To easy identify such auto-generated roles and bindings,
+		their names are prefixed by 'psp:' string.
+
+		NOTE: it does not modify anything by default.`)
+
+	internalMigrateSCCExample = templates.Examples(`
+		# Perform a dry-run of migrating all objects
+		%[1]s
+
+		# To actually perform the migration, the confirm flag must be appended
+		%[1]s --confirm`)
+)
+
+type migrateSCCOptions struct {
+	kubeClient         *kubernetes.Clientset
+	cachedRoles        *rbacv1.RoleList
+	cachedClusterRoles *rbacv1.ClusterRoleList
+	migrate.ResourceOptions
+}
+
+func NewCmdMigrateSCC(name, fullName string, f kcmdutil.Factory, _ io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &migrateSCCOptions{
```

Comment:
Look at other commands how we instantiate Options and follow the current patterns.

Reply:
Should I just wrap the options creation in a new function or would you like me to go all the way to using `genericclioptions`? I can do the latter, but I'll have to do that for the whole `migrate` module since all the packages in it somehow use the shared `migrate.ResourceOptions` type to implement their options handling. Shall I open a different PR for that?